### PR TITLE
Fix thread & main timeline partitioning logic

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/21599

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

**Comment by gsouquet** [comment]
Does this possibly fix https://github.com/vector-im/element-web/issues/21613 ?

**Comment by t3chguy** [comment]
Unlikely, that looks like thread initialisation/pagination somehow getting the wrong events from relations

File: spec/test-utils/test-utils.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+    });
+}
+
+/**
+ * Create a spy for an object and automatically spy its methods.
+ * @param {*} constr The class constructor (used with 'new')
+ * @param {string} name The name of the class
+ * @return {Object} An instantiated object with spied methods/properties.
+ */
+export function mock(constr, name: string) {
```
Comment (gsouquet) [inline spec/test-utils/test-utils.ts:47]: Missing return type

File: src/models/room.ts [hunk 2]
```diff
@@ -1562,19 +1562,68 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
...
+        }
+
+        // We've exhausted all scenarios, can safely assume that this event should live in the room timeline only
+        return {
+            shouldLiveInRoom: true,
+            shouldLiveInThread: false,
+        };
+    }
+
+    public findThreadForEvent(event?: MatrixEvent): Thread | null {
```
Comment (gsouquet) [inline src/models/room.ts:1622]: why not make the event required here? Not sure what value that brings to have it optional

**Review by gsouquet — APPROVED** [review]
LGTM 👍

**Pull request merged**